shcomplete/filtering.py

```diff
@@ -9,9 +9,36 @@ def register_shell(cls):
     return cls
 
 
-class ShellTimestamp:
+class ShellFiltering:
 
     COMMENT_RE = re.compile(r"^#")
+    PGP_MESSAGE = re.compile(r"-----BEGIN PGP MESSAGE-----")
+    HTML_DOCTYPE = re.compile(r"<\!DOCTYPE html>")
+
+    @classmethod
+    def is_too_small(cls, file_name):
+        """
+        Check if file_name contains less than 10 lines.
+        """
+        with open(file_name, "r") as f:
+            lines = f.readlines()
+            return len(lines) < 10
```

Comment:
In very small file histories, the threshold often made me accept every thing in the vocab, even if some prefixes were specific to the file. That's why I decided to remove them, I thinks I lost only 100 or 200 bytes altogether. However, maybe the max number of lines could be an argument..